src/Session.ts

```diff
@@ -1015,22 +1016,62 @@ export class Session extends DebugSession {
         }
     }
 
-    private createSourceMapFileWatcher(sourceMapRoot?: string) {
-        if (this._fileWatcher) {
-            this._fileWatcher.dispose();
-            this._fileWatcher = undefined;
+    private createSourceMapFileWatcher(localRoot: string, sourceMapRoot?: string) {
+        if (this._sourceFileWatcher) {
+            this._sourceFileWatcher.dispose();
+            this._sourceFileWatcher = undefined;
         }
-        if (sourceMapRoot) {
-            this._fileWatcher = workspace.createFileSystemWatcher('**/*.{map}', false, false, false);
-            this._fileWatcher.onDidChange(uri => {
-                this._sourceMaps.reset();
-            });
-            this._fileWatcher.onDidCreate(uri => {
-                this._sourceMaps.reset();
-            });
-            this._fileWatcher.onDidDelete(uri => {
-                this._sourceMaps.reset();
-            });
+
+        const config = workspace.getConfiguration('minecraft-debugger');
+        const reloadOnSourceChangesEnabled = config.get<boolean>('reloadOnSourceChanges.enabled');
+        if (!reloadOnSourceChangesEnabled) {
+            return;
+        }
+
+        const reloadOnSourceChangesDelay = Math.max(config.get<number>('reloadOnSourceChanges.delay') || 0, 0);
+        const reloadOnSourceChangesGlobPattern = config.get<string>('reloadOnSourceChanges.globPattern');
+        
+        // Either monitor the build output (TS->JS) by looking at .map and .js files in sourceMapRoot,
+        // or monitor .js files directly if not using TS or source maps by looking at localRoot,
+        // or monitor a specific glob pattern for all files within the workspace.
+        let globPattern: RelativePattern | undefined = undefined;
+        if (reloadOnSourceChangesGlobPattern) {
+            const workspaceFolders = workspace.workspaceFolders;
+            if (workspaceFolders && workspaceFolders.length > 0) {
+                globPattern = new RelativePattern(workspaceFolders[0].uri.fsPath || '', reloadOnSourceChangesGlobPattern);
```

Comment:
```suggestion
                globPattern = new RelativePattern(workspaceFolders[0].uri.fsPath ?? '', reloadOnSourceChangesGlobPattern);
```

I think this is what you want. If `fsPath` is already `""`, as you have it now it still needs to go to the right hand side. Fun little quirk of JS ([playground link](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBAbgQwDYwLwwOQYNwFgBQokISApgHRIgDmAFIigD6OZSnQYCUeh4EJFKnQYwA-KNbsoXbEA))